app/src/main/java/com/infomaniak/mail/ui/main/move/MoveFragment.kt

```diff
@@ -101,79 +110,90 @@ class MoveFragment : MenuFoldersFragment() {
-            }
+    private fun observeFolderCreation() = with(mainViewModel) {
 
-            defaultFoldersAdapter.setFolders(defaultFoldersWithoutDraft, folderId)
-            customFoldersAdapter.setFolders(customFolders, folderId)
-            setSearchBarUi(allFolders = defaultFoldersWithoutDraft + customFolders)
+        newFolderResultTrigger.observe(viewLifecycleOwner) {
+            inputDialog.resetLoadingAndDismiss()
         }
-    }
 
-    private fun observeNewFolderCreation() = with(mainViewModel) {
-        newFolderResultTrigger.observe(viewLifecycleOwner) { inputDialog.resetLoadingAndDismiss() }
         isMovedToNewFolder.observe(viewLifecycleOwner) { isFolderCreated ->
             if (isFolderCreated) findNavController().popBackStack()
         }
     }
 
-    override fun onFolderSelected(folderId: String): Unit = with(navigationArgs) {
+    private fun onFolderSelected(folderId: String): Unit = with(navigationArgs) {
         mainViewModel.moveThreadsOrMessageTo(folderId, threadsUids.toList(), messageUid)
         findNavController().popBackStack()
     }
 
-    override fun onFolderCollapse(folderId: String, shouldCollapse: Boolean) = Unit
+    /**
+     * Asynchronously validate folder name locally
+     * @return error string, otherwise null
+     */
+    private fun checkForFolderCreationErrors(folderName: CharSequence): String? = when {
+        folderName.length > 255 -> getString(R.string.errorNewFolderNameTooLong)
+        folderName.contains(Regex(INVALID_CHARACTERS_PATTERN)) -> getString(R.string.errorNewFolderInvalidCharacter)
+        folderController.getRootFolder(folderName.toString()) != null -> context?.getString(R.string.errorNewFolderAlreadyExists)
+        else -> null
+    }
 
-    private fun setSearchBarUi(allFolders: List<Folder>) = with(binding) {
-        searchResultsList.adapter = searchResultsAdapter
+    private fun setupSearchBar(allFolders: List<Folder>, currentFolderId: String) = with(binding) {
 
         searchInputLayout.setOnClearTextClickListener { trackMoveSearchEvent(SEARCH_DELETE_NAME) }
 
         searchTextInput.apply {
-            toggleFolderListsVisibility(!text.isNullOrBlank())
+
             doOnTextChanged { newQuery, _, _, _ ->
-                toggleFolderListsVisibility(!newQuery.isNullOrBlank())
+
                 if (newQuery?.isNotBlank() == true) {
-                    moveViewModel.filterFolders(newQuery.toString(), allFolders, shouldDebounce = true)
+                    moveViewModel.filterFolders(newQuery.toString(), allFolders, currentFolderId, shouldDebounce = true)
+                } else {
+                    moveAdapter.setFolders(allFolders, currentFolderId)
                 }
 
-                if (!hasAlreadyTrackedFolderSearch) {
+                if (!hasAlreadyTrackedSearch) {
                     trackMoveSearchEvent("executeSearch")
-                    hasAlreadyTrackedFolderSearch = true
+                    hasAlreadyTrackedSearch = true
```

Comment:
This would track again after every configuration change. Is it a problem? I don't know

Reply:
Fixed.